packages/vitest/src/typecheck/typechecker.ts

```diff
@@ -318,6 +318,20 @@ export class Typechecker {
       throwOnError: false,
     })
     this.process = child.process
+
+    const abortController = new AbortController()
+    await Promise.race([
+      new Promise(resolve => child.process?.once('spawn', resolve)),
+      new Promise((_resolve, reject) => child.process?.once('error', (cause) => {
+        reject(new Error(`Spawning typechecker failed - is typescript installed?`, { cause }))
+      })),
+      new Promise((_, reject) => {
+        const timeout = setTimeout(() => reject(new Error(`${typecheck.checker} spawn timed out`)), 60000)
+        abortController.signal.addEventListener('abort', () => clearTimeout(timeout), { once: true })
+      }),
+    ])
+      .finally(() => abortController.abort())
```

Comment:
Could this be simplified to something like this? I think `new Promise((resolve, reject) ...)` itself works like race, so splitting all cases seems unnecessary.

```ts
    await new Promise<void>((resolve, reject) => {
      const timeout = setTimeout(() => reject(new Error(`${typecheck.checker} spawn timed out`)), 60_000)
      child.process?.once('spawn', () => {
        clearTimeout(timeout);
        resolve();
      })
      child.process?.once('error', (cause) => {
        clearTimeout(timeout);
        reject(new Error(`Spawning typechecker failed - is typescript installed?`, { cause }))
      })
    })
```